Assets/AWSIM/Scripts/RandomTraffic/TrafficManager.cs

```diff
@@ -0,0 +1,205 @@
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace AWSIM.TrafficSimulation
+{
+/// <summary>
+/// Component for managing traffic simulators. Traffic manager collets all traffic simulators and manages the spawning process.
+/// - Reproducibility by Seed value
+/// </summary>
+public class TrafficManager : MonoBehaviour
+{
+    [SerializeField, Tooltip("Seed value for random generator.")]
+    private int seed;
+    [Header("NPC Vehicle Settings")]
+    [SerializeField] private NPCVehicleConfig vehicleConfig = NPCVehicleConfig.Default();
+    [SerializeField] private LayerMask vehicleLayerMask;
+    [SerializeField] private LayerMask groundLayerMask;
+    [SerializeField] public int maxVehicleCount = 40;
```

Comment:
I think it is reasonable to keep both `maximumSpawns` and `maxVehicleCount` because of a few reasons:

- With only `maximumSpawns` we can't control the number of NPC currently living in a scenario. Without the limit (`maximumSpawns = 0`), the NPCs will spawn infinitely and oversaturate the scene. There is also no mechanism to stagger the spawning of the NPCs, and even if there were - we still wouldn't be able to hold the number of vehicles reliably.
- If we set the fixed number of `maximumSpawns` and there are no any other spawn mechanisms, then after some time, we may end up with an empty scene - vehicles can despawn if there are dangling lanelets.
- It would be quite common to spawn only _N_ number of vehicles (i.e., a single Ego vehicle following the path) and simultaneously keep the overall number of vehicles no more than `maxVehicleCount`.

Let's assume an example scenario: Ego vehicle moves via the designed route in a semi-crowded area - infinitely. We, therefore, need `maximumSpawns` to spawn only one Ego, and `maxVehicleCount` to keep the "semi-crowded" number of vehicles in a scene.

What do you think :bow: ?

Reply:
Here is the discomfort I am feeling
- Users have difficulty understanding the relationship between `MaxVehicleCount` and `maximumSpawns`.
- I don't think the spec that allows infinite spawning at 0 is a good idea. It is counter-intuitive. When it is 0, shouldn't the number of spawns be 0 to begin with?

Please point out if I have missed something.

Reply:
I understand your concerns. If it comes to me, I see these two parameters as follows:

- `MaxVehicleCount` - a parameter describing the limit of vehicles in a simulation. Used for controlling the traffic density as well as a way to meet the performance limits of the machine,
- `maximumSpawns` - a parameter to describe the life-time of an individual traffic simulator instance. Setting it makes the spawner live longer or shorter, while it can also be set to infinity if needed (endless life-time).

Maybe I should put some more elaborate descriptions for each, and that could help users to distinguish these two :thinking: ? Maybe I could also change `maximumSpawns` to `lifetime` and add some description in a tooltip?

I also looked into the code, and the `MaxVehicleCount` seems heavily tied with the cognition step. The sizes of `NativeArray`s are currently initialized with the maximum number of vehicles (see [here](https://github.com/tier4/AWSIM/blob/main/Assets/AWSIM/Scripts/RandomTraffic/NPCVehicle/Steps/NPCVehicleCognitionStep.cs#L587-L598)). We could use the sum of `maximumSpawns` for each traffic simulator to set the length of that arrays, but that won't be optimal in case of adding traffic sims runtime. We would have to reinitialize them whenever a new traffic simulator is added during the runtime. Therefore, removing the `MaxVehicleCount` could have some deeper implications.

As for the `0 == infinity` - as far as I know, it is common to use `0` as a special case. Still, I can add a button for "infinity life-time" and remove the "enabled" variable which is used for temporarily turning on/off individual simulators :+1: :bow:

Reply:
Hmmm. I see. For now, let's leave the MaxVehicleCount parameter as it is. If possible, it would be great if there is an explanation in tool tips, etc. Also, it would be more helpful if there are examples of settings in the documentation.

Another point I was wondering, is it possible to keep more than one Random Traffic Sim? If not, there may be no need to have an array of RandomTrafficSims in TrafficManager.
![image](https://github.com/tier4/AWSIM/assets/14327102/46a77b6a-d396-4fbf-b213-e71b62e32c93)

Reply:
> If possible, it would be great if there is an explanation in tool tips, etc.

Yes, I will add them :+1: 

> is it possible to keep more than one Random Traffic Sim?

Yes, it is possible to add multiple random traffics. We can use it, for example, if we want to spawn just one "truck" (or any special-case vehicle) in a whole random traffic. In that case we setting up one random traffic with a "truck" npc and `masSpawns=1`, and a second random traffic with all other models, without the spawn limit.